service-search-nls/src/main/java/fi/nls/oskari/search/util/ELFGeoLocatorCountries.java

```diff
@@ -31,53 +30,118 @@
     private long lastSync;
     private String serviceURL;
     private ELFGeoLocatorSearchChannel channel;
-    private Map<String, String> countries = new HashMap<>();
+    private static Map<String, String> countryMap = new HashMap<>();
 
-    public ELFGeoLocatorCountries(ELFGeoLocatorSearchChannel channel) {
-        this.channel = channel;
+    public ELFGeoLocatorCountries() {
         serviceURL = PropertyUtil.getOptional(ELFGeoLocatorSearchChannel.PROPERTY_SERVICE_URL);
         if (serviceURL == null) {
             log.warn("ServiceURL not configured. Add property with key", ELFGeoLocatorSearchChannel.PROPERTY_SERVICE_URL);
         }
+        loadCountryMap();
     }
 
-    public Map<String, String> getCountryMap() throws IOException {
-        // not properly configured
-        if (serviceURL == null) {
-            return countries;
+    public void loadCountryMap () {
```

Comment:
Seems like a bad design. Why should this be public if it doesn't return anything?

Reply:
This refactoring was made to make unit tests possible. Couldn't think of a better solution at the moment.